rama-dns/src/in_memory.rs

```diff
@@ -59,6 +64,16 @@ impl InMemoryDns {
         self
     }
 
+    /// Convenience method to insert an IPv4 address for a domain.
+    pub fn insert_ipv4(&mut self, name: impl Into<Domain>, addr: Ipv4Addr) -> &mut Self {
```

Comment:
```suggestion
    pub fn insert_ipv4(&mut self, name: Domain, addr: Ipv4Addr) -> &mut Self {
```

There is not a whole lot that can be turned directly into a domain as an infallible process. So might as well just expect the concrete type itself.